addons/stock/models/stock_move_line.py

```diff
@@ -201,7 +201,7 @@ def _onchange_qty_done(self):
     def _onchange_putaway_location(self):
         if not self.id and self.user_has_groups('stock.group_stock_multi_locations') and self.product_id and self.qty_done:
             qty_done = self.product_uom_id._compute_quantity(self.qty_done, self.product_id.uom_id)
-            default_dest_location = self._get_default_dest_location()
+            default_dest_location = self._get_default_dest_location() if self.user_has_groups('stock.group_stock_storage_categories') else self.location_dest_id
```

Comment:
It should be in _get_default_dest_location itself IMO. So we avoid the same issue with the put in pack